user/db.go

```diff
@@ -68,273 +67,179 @@ func (s *Store) getUser(ctx context.Context, user User) (User, error) {
 	return u, nil
 }
 
-// getUsers will get users
-func (s *Store) getUsers(ctx context.Context) ([]User, error) {
+// //getUsersSizePage will get users with page size
+// func (s *Store) getUsersSizePage(ctx context.Context, size, page int) ([]User, error) {
+// getUsers will get users with page size
+func (s *Store) getUsers(ctx context.Context, size, page int, enabled, deleted string) ([]User, error) {
 	var u []User
-	err := s.db.SelectContext(ctx, &u,
-		`SELECT *
-		FROM people.users;`)
-	if err != nil {
-		return nil, err
+	enabledSQL := s.parseEnabled(enabled, false)
+	deletedSQL := s.parseDeleted(deleted, len(enabledSQL) > 0)
+	var where string
+	if len(enabledSQL) > 0 || len(deletedSQL) > 0 {
+		where = `WHERE`
 	}
-	return u, nil
-}
-
-// getUsersSizePage will get users with page size.
-// Size is specified by the users page, size of list, from 5 to all items.
-// Page is the page number displayed, this is for the UI users page
-func (s *Store) getUsersSizePage(ctx context.Context, size, page int) ([]User, error) {
-	var u []User
-	// SELECT u.*, COUNT(*) / $1 AS pages
-	err := s.db.SelectContext(ctx, &u,
-		`SELECT u.*
+	pageSize := s.parsePageSize(page, size)
+	err := s.db.SelectContext(ctx, &u, fmt.Sprintf(`SELECT u.*
 		FROM people.users u
--- 		GROUP BY u, user_id, username, university_username, email, first_name, last_name, nickname, login_type, password, salt, avatar, last_login, reset_pw, enabled, created_at, created_by, updated_at, updated_by, deleted_at, deleted_by, use_gravatar, ldap_username
-		LIMIT $1
-		OFFSET $2;`, size, size*(page-1))
+		%[1]s
+		%[2]s
+		%[3]s
+		%[4]s;`, where, enabledSQL, deletedSQL, pageSize))
```

Comment:
I tried googling but I didn't find much, guess I need to improve my googling, I'll check this tomorrow and see if I can implement this

Reply:
I'm actively working on this

Reply:
(fwiw, if you think the end result isn't an improvement over the `Sprintf` approach I'm happy for you to keep this - I just thought it might be worth a try for some of the more complex queries)

Reply:
It seems to be working just fine for the permissions and roles db.go files, yes I put it in there but does actually work quite nicely, one of them I don't think I'll be able to do `countUsersAll`

Reply:
Honestly I'd say it might not be worth it for some of the simpler `SELECT`s and `INSERT`s - I'll let you judge

Reply:
@markspolakovs I have done a few changes to the db files, mind seeing if it is better?